### ISSUE TITLE
Unstable numerics in MvNormal KL

### ISSUE THREAD

**Issue by ferrine** [tensorflow/tensorflow#13430]
https://github.com/tensorflow/tensorflow/blame/0cfb16e025b3d20e8c8aca431fc0887814817c44/tensorflow/contrib/distributions/python/ops/mvn_linear_operator.py#L302

This line should look like this:

```python
def squared_frobenius_norm(x):                                          
    """Helper to make KL calculation slightly more readable. And Stable!
    square(sqrt(sqnorm)) = sqnorm                                                 
    """                                                                 
    # http://mathworld.wolfram.com/FrobeniusNorm.html                   
    return tf.reduce_sum(x * tf.conj(x), axis=[-2, -1], keep_dims=False)
```

It yields NaN in grad KL(q||p) if q=p

**Comment by asimshankar** [comment]
@jvdillon @langmore : Mind taking a look?

**Comment by jvdillon** [comment]
Thanks for alerting me. I will take a look!

**Comment by jvdillon** [comment]
Fixed. Should hit master tomorrow or the day after.  Thanks for your suggested fix! Not sure why linalg.norm doesnt work, that seems like the real bug.

**Comment by ferrine** [comment]
It doesn't work because gradient is unstable function and graph is not simplified
